src/lightmap.cpp

```diff
@@ -340,9 +338,9 @@ void map::build_sunlight_cache( int pzlev )
                     if( prev_transparency > LIGHT_TRANSPARENCY_SOLID &&
                         !prev_floor_cache[prev_x][prev_y] &&
                         ( prev_light_max = prev_lm[prev_x][prev_y].max() ) > 0.0 ) {
-                        const float light_level = std::max( inside_light_level, prev_light_max *
+                        const float light_level = std::min( prev_light_max, std::max( inside_light_level, prev_light_max *
```

Comment:
You can use [clamp](https://github.com/CleverRaven/Cataclysm-DDA/blob/16dfef8d18a9c597a73f8c4f05c941b286514650/src/cata_utility.h#L222) here.